src/excmds.ts

```diff
@@ -400,11 +400,23 @@ export async function updatenative(interactive = true) {
  */
 //#background
 export async function restart() {
-    const firefox = (await Native.ffargs()).join(" ")
     const profile = await Native.getProfileDir()
-    // Wait for the lock to disappear, then wait a bit more, then start firefox
-    Native.run(`while readlink ${profile}/lock ; do sleep 1 ; done ; sleep 1 ; ${firefox}`)
-    qall()
+
+    if ((await browser.runtime.getPlatformInfo()).os === "win") {
+        let reply = await Native.winRestartFirefox(profile)
```

Comment:
Please add an "interactive" version gate here, `nativegate`, to whichever version you bump it to.

Reply:
[Done!][0]

[0]: https://github.com/gsbabil/tridactyl/commit/ef788a393f09c17858e10e42f61eb00f6a7ed96e#diff-9ddd9172e4a54cdd30d56e5f10d5facaR251